src/main/java/com/wanderersoftherift/wotr/item/RiftChestType.java

```diff
@@ -0,0 +1,32 @@
+package com.wanderersoftherift.wotr.item;
+
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.DataResult;
+
+public enum RiftChestType {
+
+    WOODEN("wooden");
+
+    public static final Codec<RiftChestType> CODEC = Codec.STRING.flatComapMap(s -> RiftChestType.byName(s, null), d -> DataResult.success(d.getName()));
+
+    private final String name;
+
+    RiftChestType(String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public static RiftChestType byName(String name, RiftChestType defaultReturn) {
+        for (RiftChestType value : values()) {
+            if (value.name.equalsIgnoreCase(name)) {

```

Comment:
Need a locale on the equalsIgnoreCase (probably `Locale.ROOT`)

Reply:
I mean, the possible values are the enum values, so unsure how valuable it is. But I guess in terms of clean code it is necessary.

Reply:
"WOODEN" doesn't equal "wooden" for someone from Turkey is all.

Reply:
After checking, no Locale possible on this method. Need to use a Collator. Seems a bit much for ensuring an enum value without weird characters matches...